File: aoc2023/java/src/test/java/com/peeekay/aoc2023/java/Day03Test.java
package com.peeekay.aoc2023.java;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;

import static org.junit.jupiter.api.Assertions.assertEquals;

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class Day03Test {
    final AOCPuzzle day = new Day03(true);

    @Test
    void part1() {
        day.solve();
        assertEquals(4361, day.part1());
    }

    @Test
    void part2() {
        day.solve();
        assertEquals(467835L, day.part2());
    }

}

File: aoc2024/kotlin/src/test/kotlin/Day03Test.kt
package com.peeekay.aoc2024.kotlin

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

@DisplayName("Day 03")
internal class Day03Test {

    @Test
    fun `part1 test data solve gives correct answer`() {
        val day = Day03(isTest = true)
        val answer = day.partOne()

        assertEquals(161, answer)
    }

    @Test
    fun `part2 test data solve gives correct answer`() {
        val day = Day03(isTest = true)
        val answer = day.partTwo()

        assertEquals(161, answer)
    }

}
